atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -96,84 +94,52 @@ public OracleNamespaceDeleterIntegrationTest() {
+                .from(dbKeyValueServiceConfig)
+                .ddl(ImmutableOracleDdlConfig.builder()
+                        .from(oracleDdlConfig)
+                        .useTableMapping(false)
+                        .build())
                 .build();
 
         keyValueServiceWithDefaultPrefixNoMapping =
-                ConnectionManagerAwareDbKvs.create(ImmutableDbKeyValueServiceConfig.builder()
-                        .from(dbKeyValueServiceConfig)
-                        .ddl(ddlConfigWithDefaultPrefixNoMapping)
-                        .build());
+                ConnectionManagerAwareDbKvs.create(kvsConfigWithDefaultPrefixNoMapping);
 
-        namespaceDeleterWithDefaultPrefixNoMapping = createNamespaceDeleter(
-                ddlConfigWithDefaultPrefixNoMapping,
-                dbKeyValueServiceConfig.connection(),
-                DbKvsOracleTestSuite.getConnectionSupplier(keyValueServiceWithDefaultPrefixNoMapping),
-                MoreExecutors.newDirectExecutorService());
+        namespaceDeleterWithDefaultPrefixNoMapping =
+                factory.createNamespaceDeleter(kvsConfigWithDefaultPrefixNoMapping, RUNTIME_CONFIG);
 
         timestampTableName = oracleDdlConfig.tablePrefix() + AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName();
     }
 
-    private static NamespaceDeleter createNamespaceDeleter(
-            OracleDdlConfig ddlConfig,
-            ConnectionConfig connectionConfig,
-            ConnectionSupplier connectionSupplier,
-            ExecutorService executorService) {
-        OracleTableNameGetter oracleTableNameGetter = OracleTableNameGetterImpl.createDefault(ddlConfig);
-
-        Function<TableReference, OracleDdlTable> ddlTableFactory = tableReference -> OracleDdlTable.create(
-                tableReference,
-                connectionSupplier,
-                ddlConfig,
-                oracleTableNameGetter,
-                new TableValueStyleCache(),
-                executorService);
-
-        return new OracleNamespaceDeleter(OracleNamespaceDeleterParameters.builder()
-                .tablePrefix(ddlConfig.tablePrefix())
-                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
-                .connectionSupplier(connectionSupplier)
-                .tableNameGetter(oracleTableNameGetter)
-                .oracleDdlTableFactory(ddlTableFactory)
-                .userId(connectionConfig.getDbLogin())
-                .build());
-    }
-
     @After
-    public void after() {
+    public void after() throws IOException, SQLException {
         namespaceDeleter.deleteAllDataFromNamespace();
         namespaceDeleterWithNonDefaultPrefix.deleteAllDataFromNamespace();
         namespaceDeleterWithDefaultPrefixNoMapping.deleteAllDataFromNamespace();
+        namespaceDeleter.close();
+        namespaceDeleterWithDefaultPrefixNoMapping.close();
+        namespaceDeleterWithNonDefaultPrefix.close();
```

Comment:
There's probably something here around the resources for deleters later in the sequence not getting cleaned up in the event that `close` throws an exception. I'd hope this is unlikely, but from the `throws` it's clearly something we didn't rule out

Reply:
Yeah. In this particular case, if cleaning up the namespace failed then we're already in a bad state (the next test will likely fail then cleanup). Close doesn't actually throw an exception (but the Closeable interface declares one), and since we're using a shared connection in these tests (a single connection supplier), if any of them succeed, they all technically have.